### PR TITLE
Adding Basic Test Case and CGeometry Unit Tests

### PR DESCRIPTION
**Pull request by talbring**
## Proposed Changes

This is an attempt to add some unit testing for the geometry structure that checks all routines necessary to construct it. I know that I stretch the term "unit test" a little bit here, but because of the structure of the code, its not possible to implement small, independent tests ... but this is the closest we can get to a real unit testing framework I think. With this approach it is not possible to run tests in a random order for example, but at least the order in one translation unit is fixed by default anyway.
 
I added a basic, reusable setup which is based on the box mesh reader that can be included in other tests. 

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [x] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: UnitTests/UnitQuadTestCase.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#include <string>
+
+#include "../SU2_CFD/include/solvers/CSolverFactory.hpp"
+#include "../Common/include/geometry/CPhysicalGeometry.hpp"
+#include "../SU2_CFD/include/solvers/CNSSolver.hpp"
+
+struct UnitQuadTestCase {
+
+  const std::string config_options = "SOLVER= NAVIER_STOKES\n"
+                                     "KIND_VERIFICATION_SOLUTION=MMS_NS_UNIT_QUAD\n"
```
Comment (pcarruscag) [inline UnitTests/UnitQuadTestCase.hpp:12]: This is clever :+1:

File: UnitTests/Common/geometry/CGeometry_test.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+TEST_CASE("Set bound control volume", "[Geometry]"){
+
+  TestCase->geometry->SetBoundControlVolume(TestCase->config.get(), ALLOCATE);
+
+  CHECK(TestCase->geometry->vertex[0][4]->GetNormal()[0] == -0.0625);
+  CHECK(TestCase->geometry->vertex[3][2]->GetNormal()[1] == -0.0625);
+  CHECK(TestCase->geometry->vertex[5][3]->GetNormal()[2] ==  0.03125);
+
+}
```
Comment (economon) [inline UnitTests/Common/geometry/CGeometry_test.cpp:140]: Add end lines to these files

File: UnitTests/Common/geometry/CGeometry_test.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
+#include "catch.hpp"
```
Comment (economon) [inline UnitTests/Common/geometry/CGeometry_test.cpp:1]: Add the typical file header info to these files

File: UnitTests/UnitQuadTestCase.hpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    geometry->FindNormal_Neighbor(config.get());
+    geometry->SetGlobal_to_Local_Point();
+    geometry->PreprocessP2PComms(geometry.get(),config.get());
+
+    cout.rdbuf(orig_buf);
+
+  }
+
+  ~UnitQuadTestCase(){
+    if (solver != nullptr){
```
Comment (economon) [inline UnitTests/UnitQuadTestCase.hpp:73]: this check shouldn't be necessary

File: UnitTests/UnitQuadTestCase.hpp [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      geometry = std::unique_ptr<CGeometry>(new CPhysicalGeometry(aux_geometry.get(), config.get()));
+    }
+    geometry->SetSendReceive(config.get());
+    geometry->SetBoundaries(config.get());
+    geometry->SetPoint_Connectivity();
+    geometry->SetElement_Connectivity();
+    geometry->SetBoundVolume();
+
+    geometry->SetEdges();
+    geometry->SetVertex(config.get());
```
Comment (pcarruscag) [inline UnitTests/UnitQuadTestCase.hpp:100]: Hey @talbring, you need the orientation checks (I just found this out while making some unit tests based on this code).

**Pull request merged**